src/client/mesh.cpp

```diff
@@ -154,11 +213,9 @@ void translateMesh(scene::IMesh *mesh, v3f vec)
 	u32 mc = mesh->getMeshBufferCount();
 	for (u32 j = 0; j < mc; j++) {
 		scene::IMeshBuffer *buf = mesh->getMeshBuffer(j);
-		const u32 stride = getVertexPitchFromType(buf->getVertexType());
-		u32 vertex_count = buf->getVertexCount();
-		u8 *vertices = (u8 *)buf->getVertices();
-		for (u32 i = 0; i < vertex_count; i++)
-			((video::S3DVertex *)(vertices + i * stride))->Pos += vec;
+		video::S3DVertex *vertices = (video::S3DVertexTangents*)buf->getVertices();
```

Comment:
This looks wrong, should be `S3DVertexTangents *vertices` (or just use `auto`).

ps. use `static_cast` for casting void pointers

pps. not a full review; is this still WIP (I'm seeing outcommented code etc.)?

Reply:
Actually it is already in the ready-for-review state. I forgot about the outcommented code of adding the emissive color for the "wielditem" luaobject type (it is unnecessary since the wieldmeshes use nodes_shader instead of object one now), also setMeshTangentsColor is not used anywhere.

Reply:
I've replaced it to `static_cast` throughout the `mesh.cpp` file and partially in `mapblock_mesh.cpp`. Also, I mistook concerning to the emissive color: it is added for `m_meshnode`, not `m_wieldmesh_node` in `GenericCAO::setNodeLight`, so I reverted it.